components/list/list.js

```diff
@@ -91,17 +96,36 @@ class List extends SelectionMixin(LitElement) {
 		`;
 	}
 
+	getItems() {
+		const slot = this.shadowRoot.querySelector('slot:not([name])');
+		if (!slot) return [];
+		return slot.assignedNodes({ flatten: true }).filter((node) => {
+			return node.nodeType === Node.ELEMENT_NODE && (node.role === 'rowgroup' || node.role === 'listitem');
+		});
+	}
+
+	getListItemByKey(key) {
+		const items = this.getItems();
+		for (let i = 0; i < items.length; i++) {
+			if (items[i].key === key) return items[i];
+			if (items[i]._selectionProvider) {
+				const tempItem = items[i]._selectionProvider.getListItemByKey(key);
+				if (tempItem) return tempItem;
+			}
+		}
```

Comment:
This will return `undefined` if the item isn't found -- is that what you'd like, or `null`?

Reply:
I don't have a strong opinion either way.  Do you have a preference?

Reply:
Not really, I think we're inconsistent. Mostly I like to return _something_ (or throw if we're not expecting someone to ever _not_ get something) just so it's clear that I actually remembered this possibility.